api-compose-object.go

```diff
@@ -55,7 +61,7 @@ type DestinationInfo struct {
 // if needed. If nil is passed, and if only a single source (of any
 // size) is provided in the ComposeObject call, then metadata from the
 // source is copied to the destination.
-func NewDestinationInfo(bucket, object string, sse encrypt.ServerSide, userMeta map[string]string) (d DestinationInfo, err error) {
+func NewDestinationInfo(bucket, object string, sse encrypt.ServerSide, userMeta, userTags map[string]string, replaceTags bool) (d DestinationInfo, err error) {
```

Comment:
might be better to have a NewDestinationInfoWithOptions for future extensibility